### PR TITLE
chore: support multi tag value selection

### PR DESCRIPTION
**Pull request by appletreeisyellow**
Close influxdata/ui#5949

This PR updates the AST for multi tag values in the following rules:

* If adding a tag value where the tag key already exists in the flux file, use `or`.
* If adding a tag value with a new tag key, append a new `filter(...)` function. This function as implicit `and`.

### CODE REVIEW

**Comment by appletreeisyellow** [comment]
Good question! I'll add test cases for queries with `and` in the `filter()` function and see the result

File: src/composition.rs [hunk 1]
```diff
@@ -326,7 +328,8 @@ struct CompositionQueryAnalyzer {
     bucket: String,
     measurement: Option<String>,
     fields: Vec<String>,
-    tag_values: Vec<(String, String)>, // (TagName, TagValue)
+    tag_values: HashMap<String, Vec<String>>, // [(TagName, [TagValue1s])]
+    tag_keys_order: Vec<String>, // used to keep track of the order for tag keys
```
[the remark this answers is not in the document]
Comment (appletreeisyellow) [reply src/composition.rs:332]: Ah, I was looking for something like ordered_map

File: src/composition.rs [hunk 2]
```diff
@@ -444,7 +449,19 @@ impl<'a> ast::walk::Visitor<'a> for CompositionQueryAnalyzer {
...
+                                            self.tag_values[i].1.push(string_literal.value.clone());
+                                            inserted = true;
+                                            break;
+                                        }
+                                    }
+                                    if !inserted {
+                                        self.tag_values.push((
+                                            ident.name.clone(), vec![string_literal.value.clone()]
+                                        ));
+                                    }
```
Comment (appletreeisyellow) [inline src/composition.rs:464]: I have this pattern through out this file to set/get `self.tag_values`. The reason I chose to use a `for` loop and have a `break` in the loop is to save the number of iterations we have to do on a vec. In this code, there is only one iteration.

I'm new to rust so there might be functions/patterns do the same thing more efficiently and easier to read. I'm open to change to other ways!

**Comment by appletreeisyellow** [comment]
> I'm curious to know if this breaks the sync for existing queries

@asalem1 I added more test cases to ensure this change doesn't break the existing sync

**Review by rockstar — APPROVED** [review]
I suspect this is going to cause _huge_ conflicts with my work in flight, but that's on me, not on you. Thanks for getting to this!

**Pull request merged**